include/gatekeeper_main.h

```diff
@@ -47,11 +41,14 @@ extern int gatekeeper_logtype;
 	} while (0)
 
 #define G_LOG(level, ...)		        \
-	rte_log_ratelimit(RTE_LOG_ ## level,	\
-	gatekeeper_logtype, "GATEKEEPER: " __VA_ARGS__)
+	do {								\
+		rte_log_set_level(BLOCK_LOGTYPE, RTE_LOG_DEBUG);	\
+		rte_log_ratelimit(RTE_LOG_ ## level,			\
+		BLOCK_LOGTYPE, "GATEKEEPER: " __VA_ARGS__);		\
+	} while (0)
 
 #define G_LOG_CHECK(level)	\
-	check_log_allowed(RTE_LOG_ ## level, gatekeeper_logtype)
+	check_log_allowed(RTE_LOG_ ## level, BLOCK_LOGTYPE)
```

Comment:
`check_log_allowed()` must be adjusted to check the passed log level against the log level of the block instead of the log level of `BLOCK_LOGTYPE`. Otherwise, the return is always going to be true.